modules/redtramBidAdapter.js

```diff
@@ -0,0 +1,158 @@
+import {
+  isFn,
+  isStr,
+  deepAccess,
+  getWindowTop,
+  triggerPixel
+} from '../src/utils.js';
+import { registerBidder } from '../src/adapters/bidderFactory.js';
+import { BANNER } from '../src/mediaTypes.js';
+import { config } from '../src/config.js';
+import { convertOrtbRequestToProprietaryNative } from '../src/native.js';
+
+const BIDDER_CODE = 'redtram';
+const AD_URL = 'https://prebid.redtram.com/pbjs';
+const SYNC_URL = 'https://prebid.redtram.com/sync';
+
+function isBidResponseValid(bid) {
+  if (!bid.requestId || !bid.cpm || !bid.creativeId ||
+    !bid.ttl || !bid.currency || !bid.meta) {
+    return false;
+  }
+
+  switch (bid.mediaType) {
+    case BANNER:
+      return Boolean(bid.width && bid.height && bid.ad);
+    default:
+      return false;
+  }
+}
+
+function getBidFloor(bid) {
+  if (!isFn(bid.getFloor)) {
+    return deepAccess(bid, 'params.bidFloor', 0);
+  }
+
+  try {
+    const bidFloor = bid.getFloor({
+      currency: 'USD',
+      mediaType: '*',
+      size: '*',
+    });
+    return bidFloor.floor;
+  } catch (_) {
+    return 0
+  }
+}
+
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [BANNER],
+
+  isBidRequestValid: (bid) => {
+    return Boolean(bid.bidId && bid.params && bid.params.placementId);
+  },
+
+  buildRequests: (validBidRequests = [], bidderRequest) => {
+    validBidRequests = convertOrtbRequestToProprietaryNative(validBidRequests);
```

Comment:
I'm a little nervous about building to a legacy format from the jump. What's the goal here?

@dgirardi can you remind me if we plan long term support for this function. I seem to recall deprecation discussions

Reply:
In this case it looks unnecessary, since the only supported media type is banner. 
The function call is useful only if you want to get native adUnits in the "legacy" prebid format.

Reply:
Ok, I understand you. Removed this function from the code.